systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -788,23 +813,17 @@ void testDynamicallySetConnectLoggingLevels() {
             .build();
 
         LOGGER.info("Setting log level of Connect to OFF");
+
         // change to the external logging
         KafkaConnectResource.replaceKafkaConnectResourceInSpecificNamespace(testStorage.getClusterName(), conn -> {
             conn.getSpec().setLogging(connectXternalLogging);
         }, testStorage.getNamespaceName());
 
-        TestUtils.waitFor("Logger change", TestConstants.GLOBAL_POLL_INTERVAL, TestConstants.GLOBAL_TIMEOUT,
-            () -> cmdKubeClient().namespace(testStorage.getNamespaceName()).execInPod(scraperPodName, "curl", "http://" + KafkaConnectResources.serviceName(testStorage.getClusterName())
-                + ":8083/admin/loggers/root").out().contains("OFF")
-        );
-
-        TestUtils.waitFor("log to be empty", Duration.ofMillis(100).toMillis(), TestConstants.SAFETY_RECONCILIATION_INTERVAL,
-            () -> {
-                String kcLog = StUtils.getLogFromPodByTime(testStorage.getNamespaceName(), connectPodName, "", "30s");
-                return kcLog != null && kcLog.isEmpty() && !DEFAULT_LOG4J_PATTERN.matcher(kcLog).find();
-            });
+        // check if there are no more new lines in Logs
+        final Predicate<String> hasLogLevelOff = connectLogs -> connectLogs != null && connectLogs.isEmpty();
+        KafkaConnectUtils.waitForConnectLogLevelChangePropagation(testStorage, connectPods, scraperPodName, hasLogLevelOff, "OFF");
 
-        assertThat("Connect Pod should not roll", PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaConnectSelector()), equalTo(connectSnapshot));
+        assertThat("Connect Pod should not roll", PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaConnectSelector()), equalTo(connectPods));
```

Comment:
in current situation this catch problem (of unwanted roll) too, but yeah I guess the beauty of your suggestion is in fact that it would caught problem sooner. Will do thanks.